### DISCUSSION TITLE
0.8.1

### DISCUSSION THREAD

**Discussion by Larian77** [Larian77/Openbiblio#55] [General]
@375michael40veit 
Finde bei Github keine Möglichkeit Direktnachrichten zu schreiben :-) Da ich die zwei Wochen noch sehr viel zu tun habe.
Kannst du bitte mal noch die Pull Requests checken, da sind paar Bugfixes dabei, ob wir die noch in die 0.8.1 reinpacken bevor wir das offizielle File für die neue Version bereitstellen? 

Danach würde ich noch ein offizielles 0.8.1-File bereitstellen und es auch auf openbiblio.de noch hochladen :)

**Comment by 375michael40veit** [comment]
@Larian77 
Ich bin gerade dabei. Kämpfe zum Teil noch mit github, aber sollte Schritt für Schritt gehen.
Inhaltlich gibt es für noch zwei Fragen zu den bestehenden Pull Requests und zum Issue "Zurückspringen aus einer Deitailbuchanzeige in das vorherige Ergebnis wirft Fehler ab".

Hier bist Du jeweils extra mit im "Verteiler".

**Reply by Larian77** [reply]
@375michael40veit vielen Dank schon mal :-) -- ich habe noch eine Woche Ferien wo ich im Urlaub bin, danach hoffe ich auch wieder mehr Zeit zu haben (also ab 21.10.)
Die beiden Punke schaue ich mir gleich mal an, habe mal zur Diskussion zum mbrshipend noch erklärt, wie das Feld damals gedacht war, damit dort nichts in die falsche Richtung läuft. In der PHP8-Version ursprünglich von sarangt hatte er ne Funktion, die das Feld immer auf den Wert 1900 setzte, wenn es leer war, was zu Problemen in der ursprünglichen Funktion führte :-)

**Reply by 375michael40veit** [reply]
@Larian77 
Ich bin beruflich zurzeit sehr eingespannt, hoffe, dass in zwei Wochen wieder mehr Luft bei mir ist. Vielen Dank fürs durchchecken!

**Comment by Larian77** [comment]
Hallo @375michael40veit 

So, jetzt bin ich endlich dazu gekommen mir alles mal anzuschauen. Habe noch um die Passwort vergessen-Funktion 2 kleine Bugs gefunden (Staff ging nicht, bei den Vergessen-URLS fehlte der Pfad wenn es einen gibt).

Was mir auch aufgefallen ist, wenn in der Benutzersuche für eine Usernummer mehrere Ergebnisse kommen (https://openbiblio.de/openbiblio mit tester/test und nach der Benutzernummer 101 suchen) dann funktioniert bei denen dann das Nutzerlogin nicht, weil die Suchfunktion für den Login dann auch 4 Ergebnisse hat und dementsprechend auf Fehler geht) - ich sehe das gerade aber nicht als Riesen-Problem, aber vielleicht kann man hier die Suche irgendwann mal noch verbessern, dass sie nur nach der einen Nummer sucht und nicht nach Ähnlichen.

Ich würde mal noch die nächste Woche schauen ob ich noch was finde und dann noch die Anleitung aktualisieren und dann die 0.8.1 veröffentlichen.

**Comment by 375michael40veit** [comment]
Hallo @Larian77 
Der erste Fehler ist leicht nachzuvollziehen: falsche Variable.
Die Erstellung der URL nicht ganz so. Denn wenn ich Deine Variante verwende, dann kommt bei mir der Anhang doppelt heraus, also wie folgt: https://openbiblio.ausbildung-heilerziehungspflege.de/admin/staff_pwd_forget.php/admin/staff_pwd_newset_form.php?username=tester1&code=5c56504d0cc81c783afe98c49e1b3b86
Evtl. abhängig, ob Domain oder Subdomain?

**Reply by Larian77** [reply]
> Hallo @Larian77 Der erste Fehler ist leicht nachzuvollziehen: falsche Variable. Die Erstellung der URL nicht ganz so. Denn wenn ich Deine Variante verwende, dann kommt bei mir der Anhang doppelt heraus, also wie folgt: https://openbiblio.ausbildung-heilerziehungspflege.de/admin/staff_pwd_forget.php/admin/staff_pwd_newset_form.php?username=tester1&code=5c56504d0cc81c783afe98c49e1b3b86 Evtl. abhängig, ob Domain oder Subdomain?

Das Problem tritt glaube ich auf, sobald Openbiblio.de in einem Unterverzeichnis ist. Meine Test-Instanz ist unter https://openbiblio.de/openbiblio/ und wenn ich aus dem Opac für einen User-Account den PW vergessen-Link nutze hat es dann mit deiner Version nicht mehr geklappt, komischerweise aber mit meiner schon

Schaue gerne mal drauf ich habe dort jetzt das Ganze mit dem PW vergessen und dem Userlogin wieder eingeschaltet zum testen.

**Reply by 375michael40veit** [reply]
The bug is now fixed

**Comment by 375michael40veit** [comment]
Hallo @Larian77,

Folgende Anmerkung von Dir konnte ich nicht reproduzieren: "Was mir auch aufgefallen ist, wenn in der Benutzersuche für eine Usernummer mehrere Ergebnisse kommen (https://openbiblio.de/openbiblio mit tester/test und nach der Benutzernummer 101 suchen) dann funktioniert bei denen dann das Nutzerlogin nicht, weil die Suchfunktion für den Login dann auch 4 Ergebnisse hat und dementsprechend auf Fehler geht) - ich sehe das gerade aber nicht als Riesen-Problem, aber vielleicht kann man hier die Suche irgendwann mal noch verbessern, dass sie nur nach der einen Nummer sucht und nicht nach Ähnlichen."

Kannst Du dies noch genauer beschreiben? Wenn ich unter Passwort-Vergessen nur einen Teil des Benutzernamens eingebe, welcher auch bei anderen vorkommt, wird weder bei dem einen, noch bei dem anderen ein Passwort-Vergessen-Code angelegt und damit auch nicht versendet. Alle Nutzer-Accounts sind weiterhin über die "alten" Kennwörter erreichbar.

**Reply by Larian77** [reply]
Auch hier gerne mal auf https://openbiblio.de/openbiblio schauen. Es gibt einen User mit der Nummer 101 und einem hinterlegten Passwort, es gibt aber keine Chance dass sich der User mit diesem PW einloggen kann.

**Comment by 375michael40veit** [comment]
@Larian77 
Zu Deiner Feststellung: "Was mir auch aufgefallen ist, wenn in der Benutzersuche für eine Usernummer mehrere Ergebnisse kommen (https://openbiblio.de/openbiblio mit tester/test und nach der Benutzernummer 101 suchen) dann funktioniert bei denen dann das Nutzerlogin nicht, weil die Suchfunktion für den Login dann auch 4 Ergebnisse hat und dementsprechend auf Fehler geht) - ich sehe das gerade aber nicht als Riesen-Problem, aber vielleicht kann man hier die Suche irgendwann mal noch verbessern, dass sie nur nach der einen Nummer sucht und nicht nach Ähnlichen."

Der "Fehler" liegt in der opac/login.php-Datei in Zeile 44, bedingt durch die Funktion execSearch, welche sowohl beim Login, als auch bei der Mitgliedersuche verwendet wird. Dies macht bei der Mitgliedersuche schon Sinn, jedoch nicht beim Login. Aufgrund dessen ist eine "schnelle" Anpassung der Sql-Abfrage nicht zielführend. Ich werde in nächster Zeit mal sehen, ob man den Code in der Login ändern und ggf. eine neue Funktion in MemberQuery erstellt.
Diese Programmierung war schon in den früheren Versionen (z.B. 0.7.3g). 
____________________________________________-

Regarding your statement: “What I also noticed is that if several results come up in the user search for a user number (https://openbiblio.de/openbiblio with tester/test and search for user number 101) then the user login does not work for them, because the search function for the login then also has 4 results and accordingly goes to error) - but I don't see this as a huge problem right now, but maybe you can improve the search here at some point so that it only searches for the one number and not for similar ones.”

The “error” is in the opac/login.php file in line 44, caused by the execSearch function, which is used for both the login and the member search. This makes sense for the member search, but not for the login. Because of this, a “quick” adjustment of the sql query is not expedient. I will see in the near future whether the code in the login can be changed and, if necessary, a new function created in MemberQuery.
This programming was already in the earlier versions (e.g. 0.7.3g).

**Reply by 375michael40veit** [reply]
This bug is already fixed.

**Comment by Larian77** [comment]
@375michael40veit danke dir, das Problem mit dem Login sehe ich jetzt auch nicht als hohe Priorität, weil man ja eigentlich die Benutzernummer so wählen sollte, dass bei Eingabe der korrekten Nummer keine mehrfachen Ergebnisse kommen sollten (bei einer vierstelligen Nummer verwendet man ja führende 0 z.B. :-)

Danke für all die fixes etc von dir die letzten Tage, ich würde jetzt mal die aktuelle Version noch auf meiner Testversion drüberlegen und schauen ob alles geht, wenn es keine großen Bugs mehr gibt, würde ich dann die 0.8.1 gerne bis Weihnachten offiziell releasen. Hast du noch irgendeine Funktion, die vorher noch unbedingt mit rein muss oder lassen wir abgesehen von eventuellen Bugfixes jetzt alles so, wie es sein soll? :-)

_____________________________________________

thank you, I do not see the problem with the login as a high priority, because you should choose the user number so that when entering the correct number no multiple results should come (with a four-digit number you use leading 0 e.g. :-)

Thanks for all the fixes etc from you the last days, I would now put the current version on my test version and see if everything works, if there were no more big bugs, I would like to release the 0.8.1 officially until Christmas. Do you still have any functions that absolutely have to be included beforehand, or do we leave everything as it should be, apart from any bug fixes? :-)

**Comment by Larian77** [comment]
@375michael40veit gerade mal alle Optionen durchgespielt mit dem PW und kein Problem gefunden, Anzeige des Namens funktioniert auch 
Ich werde wohl die aktuelle Version jetzt mal bei uns aufspielen, da haben wir noch nen Produktiv-Test bis Weihnachten oder Ende des Jahres

**Comment by 375michael40veit** [comment]
Nein, wegen mir muss für 0.8.1 keine weitere Funktion mit hinein.
______________________________________________________________________________
No, I don't need to add another function for 0.8.1.

**Comment by Larian77** [comment]
Super, habe die beiden Fixes von gestern noch in meine Beta geladen und würde es jetzt mal noch bis nächste Woche laufen lassen und wenn da nichts kommt die 0.8.1 releasen :-)
______________________________________________________________________________
Great, I loaded the two fixes from yesterday into my beta and would now let it run until next week and if nothing comes, release 0.8.1 :-)

**Comment by Larian77** [comment]
Ich habe nun die 0.8.1. released, weil es bei uns 3 Wochen problemlos produktiv lief. 
Wollte die Anleitungen noch aktualisieren, aber sag dass @375michael40veit das ja auch schon erledigt hatte - danke :-)

Werde die nächste Woche noch die openbiblio.de-Seite aktualisieren.
@375michael40veit kannst du mal schauen, dass wir die letzten Änderungen noch in den 0.8.1-Branch schieben damit der auf dem gleichen Stand ist, danach können wir den "schließen" und die nächsten Änderungen erstmal wieder in Main machen

@sarangtc ich gebe Bescheid, wenn an Main wieder geändert werden kann, weil du ja auch Ideen hattest

Danke an euch alle :-)

___

I have now released the 0.8.1. because it ran productively for 3 weeks without any problems. 
I wanted to update the instructions, but I was told that @375michael40veit had already done that - thanks :-)

I will update the openbiblio.de page next week.
@375michael40veit can you make sure that we move the last changes to the 0.8.1 branch so that it is at the same level, then we can “close” it and make the next changes in Main again for now

@sarangtc I'll let you know when Main can be changed again, because you also had ideas

Thanks to all of you :-)

**Comment by Larian77** [comment]
So, die Webseite ist nun wieder aktuell und auch der 0.8.1-Branch entspricht dem main-Branch ist ist damit abgeschlossen :-)
@375michael40veit, @sarangtc, @reinhard-mueller vielen Dank für eure Funktionen und Bugfixes für die 0.8.1 und wir sind damit auch wieder offen für neue Funktionen etc. 
Außerdem noch ein Frohes Neues Jahr allen :-)
____
So, the website is now up to date again and the 0.8.1 branch corresponds to the main branch is now complete :-)
@375michael40veit, @sarangtc, @reinhard-mueller thank you very much for your functions and bugfixes for the 0.8.1 and we are now open for new functions etc. again. 
Also a Happy New Year to all :-)
